### PR TITLE
Simplify most of flatten

### PR DESCRIPTION
**Pull request by DarkDimius**
### Test plan

See included automated tests.

### CODE REVIEW

**Comment by DarkDimius** [comment]
Most of this PR is removing special-casing that shouldn't have existed in the first place

File: class_flatten/class_flatten.cc [hunk 1]
```diff
@@ -244,18 +156,17 @@ class FlattenWalk {
     // `methodScopes.stack`, which we push onto in the `preTransform* hook, and
     // pop from in the `postTransform` hook.
 
-    vector<Methods> methodScopes;
+    // vector<Methods> methodScopes;
```
Comment (DarkDimius) [inline class_flatten/class_flatten.cc]: /me delete this

File: class_flatten/BUILD [hunk 2]
```diff
@@ -1,5 +1,5 @@
 cc_library(
-    name = "flattener",
+    name = "class_flatten",
```
Comment (jez) [inline class_flatten/BUILD]: The Paul in me is worried that this doesn't end in `er`. But up to you.

**Comment by jez** [comment]
These changes landed in #2642

**Pull request closed without merging**